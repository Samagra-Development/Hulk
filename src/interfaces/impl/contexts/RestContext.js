"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class RestContext {
    constructor(host, method, protocol, path, headers) {
        this.host = host;
        this.method = method;
        this.protocol = protocol;
        this.path = path;
        this.headers = headers;
    }
    update(key, value) {
        this[key] = value;
    }
    clear(key) {
        delete this[key];
    }
    get(key) {
        return this[key];
    }
}
exports.default = RestContext;
//# sourceMappingURL=RestContext.js.map