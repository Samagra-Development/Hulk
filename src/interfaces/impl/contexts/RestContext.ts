import { Method } from "got/dist/source/as-promise/types";

export default class RestContext implements IContext {
    host: string;
    method: Method;
    protocol: string;
    path: string;
    queryParams: Map<String, String>;
    headers: Map<String, String>;

    constructor(host: string, method: Method, protocol: string, path: string, headers: Map<String, String>) {
        this.host = host;
        this.method = method;
        this.protocol = protocol;
        this.path = path;
        this.headers = headers;
    }
    update(key: string, value: any) {
        this[key] = value;
    }
    clear(key: string) {
        delete this[key];
    }

    get(key: string) {
        return this[key];
    }
}