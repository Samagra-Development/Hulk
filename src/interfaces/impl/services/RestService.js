"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const RestServiceResolver_1 = require("../serviceResolvers/RestServiceResolver");
class RestService {
    constructor(config) {
        this.resolver = new RestServiceResolver_1.default();
        this.config = config;
    }
    resolve(context) {
        return this.resolver.resolve(context, this.config.cadence);
    }
    validate() {
        throw new Error("Method not implemented.");
    }
}
exports.default = RestService;
//# sourceMappingURL=RestService.js.map