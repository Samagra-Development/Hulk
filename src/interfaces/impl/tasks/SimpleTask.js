"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class SimpleTask {
    constructor(service, context) {
        this.service = service;
        this.context = context;
    }
    resolve() {
        return this.service.resolve(this.context);
    }
}
exports.default = SimpleTask;
//# sourceMappingURL=SimpleTask.js.map