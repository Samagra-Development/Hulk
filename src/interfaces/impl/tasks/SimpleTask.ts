export default class SimpleTask implements ITask {
    service: IService;
    context: IContext;

    constructor(service: IService, context: IContext) {
        this.service = service;
        this.context = context;
    }
    
    resolve() {
        return this.service.resolve(this.context);
    }
}