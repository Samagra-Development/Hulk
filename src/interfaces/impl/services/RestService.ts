import RestContext from "../contexts/RestContext";
import RestServiceResolver from "../serviceResolvers/RestServiceResolver";

export default class RestService implements IService {
    resolver: IServiceResolver;
    config: ServiceConfig;
    constructor(config: ServiceConfig) {
        this.resolver = new RestServiceResolver();
        this.config = config;
    }
    resolve(context: RestContext) {
      return this.resolver.resolve(context, this.config.cadence);
    }
    validate() {
        throw new Error("Method not implemented.");
    }
    
}