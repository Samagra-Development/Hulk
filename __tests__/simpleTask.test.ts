import RestContext from "../src/interfaces/impl/contexts/RestContext";
import RestService from "../src/interfaces/impl/services/RestService"
import SimpleTask from "../src/interfaces/impl/tasks/SimpleTask";

describe("simple task test", () => {
    test("should execute", async (done) => {
        const config = {
            cadence: {
                retries: 1,
                timeout: 1000,
                concurrent: false,
                linearBackoff: 100
            }
        }
        const context: RestContext = new RestContext(
            "jsonplaceholder.typicode.com",
            "GET",
            "https",
            "/todos/200",
            new Map()
        );
        const service = new RestService(config);
        const task = new SimpleTask(service, context);
        const results = await task.resolve();
        expect(results).toMatchSnapshot();
        done();
    })
})