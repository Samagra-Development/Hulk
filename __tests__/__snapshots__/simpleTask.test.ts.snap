// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple task test should execute 1`] = `
"{
  \\"userId\\": 10,
  \\"id\\": 200,
  \\"title\\": \\"ipsam aperiam voluptates qui\\",
  \\"completed\\": false
}"
`;
